src/main/java/spark/ExceptionHandlerImpl.java

```diff
@@ -45,7 +45,7 @@ public ExceptionHandlerImpl(Class<? extends Exception> exceptionClass) {
      *
      * @param exceptionClass Type of exception
      */
-    public void exceptionClass(Class<? extends Exception> exceptionClass) {
+    public void exceptionClass(Class<T> exceptionClass) {
```

Comment:
Without the `? extends Exception` this setter has become effectively useless, as you can only set the exception type to the same type of exception. If we want to keep that behaviour, we should set the fields back to `? extends Exception` (as in my first commit). The generic type then specifies the most general type for this `ExceptionHandlerImpl` (and default to a default `Exception`), but the actual class may be more specific and set through the constructor or setter.